worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java

```diff
@@ -143,11 +143,24 @@ private TaskMessage createResponseMessage(final boolean includeTaskContext, fina
 
         final String responseMessageType = response.getMessageType();
 
+        //  Check if the 'trackTo' field needs to be reset.
+        final TrackingInfo trackingInfo;
+        if (response.getResetTrackTo()) {
+            //  Reset trackTo field to null;
+            trackingInfo = taskMessage.getTracking();
+            if (trackingInfo != null) {
+                trackingInfo.setTrackTo(null);
```

Comment:
My previous comment was to use the copy constructor and then call the `setTrackTo()` mutator - you are just using the mutator without calling the object itself.  The copy constructor is here: https://github.com/WorkerFramework/worker-framework/blob/9548a0ed0a2963fd3805430dff5277644c43cb5b/worker-api/src/main/java/com/hpe/caf/api/worker/TrackingInfo.java#L80.